### DISCUSSION TITLE
How to calculate which part (top/bottom) of `over` the `active` was when `DragEndEvent` was fired?

### DISCUSSION THREAD

**Discussion by AdsonCicilioti** [clauderic/dnd-kit#1151] [Q&A]
I have a static list of Draggable components, where I drop it on top of a Droppable area that creates a Sortable list.

 In general, when I drop a Draggable over an item in the Sortable List, in a method that captures the `DrangEndEvent` event, how can I calculate in which part of the `over` the `active` was?

 I ask this because I noticed several data with the `delta` and also the `activatorEvent.client{X,Y}`...
 I believe that an explanation of this data would be very useful for us to build our logic.

**Comment by pstrassmann** [comment]
Do you need this information after a collision has happened (in onDragOver, onDragEnd, etc) or in order to determine **if** a collision has happened?

If you want this information in order to determine collision detection, you can write a custom collision detection algorithm ([docs](https://docs.dndkit.com/api-documentation/context-provider/collision-detection-algorithms#building-custom-collision-detection-algorithms)). See the type [here](https://github.com/clauderic/dnd-kit/blob/5c58f0fe5d19b5aaa5cf93572f3435f4a0a6e54f/packages/core/src/utilities/algorithms/types.ts#L17) as well, and an [example collision detection algorithm](https://github.com/clauderic/dnd-kit/blob/5c58f0fe5d19b5aaa5cf93572f3435f4a0a6e54f/packages/core/src/utilities/algorithms/rectIntersection.ts#L38) within dndkit.

Collision detection algorithms are passed [props](https://github.com/clauderic/dnd-kit/blob/5c58f0fe5d19b5aaa5cf93572f3435f4a0a6e54f/packages/core/src/utilities/algorithms/types.ts#L17) that will allow you to get the ClientRect of the thing being dragged as well as the rectangles of all dropables. From there you can do things like calculate what parts of the draggable and droppable are intersecting, by how much, etc.
